### DISCUSSION TITLE
Does device having artpec-7 with MLPU runs Mobilenet SSD faster than that on just device's CPU ?

### DISCUSSION THREAD

**Discussion by AmarRobo** [AxisCommunications/discussions#162] [General] [answered]
I am trying to run YOLO and Mobilenet SSD on P1455 camera. Does the MPLU capability mean an edge TPU performance ? Will it have better performances than just CPU ?. In [https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/object-detection](url) example, for this camera can I select edgeTPU options ?

**Comment by daniel-falk** [comment, marked as answer]
Hi @amararjun 

The MLPU is a very narrow accelerator specialized for some specific tasks. It is not officially documented if and what operations can be scheduled to the MLPU but I would deem it unlikely that it will be of any use to you. Running a YOLO network on an A7 MLPU camera will does take many seconds per image.

The ARTPEC-7 chips with the DLPU does on the other hand have the Edge TPU which can significantly improve the inference speed of the network. Do however note that there are multiple ops that are not supported on the Edge TPU, like leaky ReLU, that is used by YOLO. Using these ops will cause the inference to be scheduled to the CPU from that point on making it potentially even slower to use the Edge TPU than the CPU due to the overhead of copying data. There are architectures like modified tiny-YOLO networks that can solve this (look at the graph produced by the compiler to verify that everything is a custom op).

**Reply by AmarRobo** [reply]
Thank you for detailed reply!

**Comment by Corallo** [comment]
Hello @amararjun 

The answer of @daniel-falk is correct, I just want to confirm that we don't support the MLPU for acap development, i.e. you can't use it to run your model.
To accelerate your model, you would need to use an Artpec-7 or Artpec-8 device with DLPU

**Reply by AmarRobo** [reply]
Thank you!
